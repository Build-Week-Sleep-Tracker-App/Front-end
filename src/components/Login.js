import React from "react";
import { withFormik, Form, Field, yupToFormErrors } from "formik";
import * as Yup from 'yup';

function Login({ errors, touched}) {
  return (
    <div>
      <h1>Login</h1>
      <Form>
        <Field type="text" name="username" placeholder="Username" />
        <Field type="password" name="password" placeholder="Password" />
        <button type="submit">Login</button>
		{touched.username && errors.username && <p>{errors.username}</p>}
		{touched.password && errors.password && <p>{errors.password}</p>}
      </Form>
    </div>
  );
}

const FormikLoginForm = withFormik({
  mapPropsToValues({ username, password, confirmPassword}) {
    return {
      username: username || "",
	  password: password || "",
    };
  },

  validationSchema: Yup.object().shape({
	  username: Yup.string().required("Please enter your username"),
	  password: Yup.string().required("Please enter your password"),
  }),	

  handleSubmit(values, { props, resetForm }) {
    props.login(values, props.history);
    resetForm();
  }
})(Login);

export default FormikLoginForm;
