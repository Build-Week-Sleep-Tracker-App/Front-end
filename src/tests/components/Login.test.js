import React from 'react';
import { shallow } from '../../enzyme';
import Login from '../../components/Login';
import { exportAllDeclaration } from '@babel/types';

const mockOnSubmit = jest.fn();
const wrapper = shallow(<Login onSubmit={mockOnSubmit} />);

describe('Login component', () => {
  test('should render login page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
