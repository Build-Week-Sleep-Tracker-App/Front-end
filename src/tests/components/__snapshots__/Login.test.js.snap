// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render login page correctly 1`] = `ShallowWrapper {}`;
